File: tests/verify/managed/fixtures/poisoned/poison.ts
// poison.ts — poisoned TypeScript fixture for the zero-alloc linter tests.
// Every commented rule id MUST fire; anything else is an engine regression.
// Module-scope allocations below exist to prove hot-region scoping.

const RAW = '{"a":1}';
const ROWS = [1, 2, 3];

/** @hot */
export function onMarketTick(tick: number, out: Float64Array): void {
  const scratch = new Float64Array(64); // expect: WV-TS-001
  const closure = (x: number) => x + tick; // expect: WV-TS-005
  const snapshot = { bid: tick, ask: tick }; // expect: WV-TS-002
  const row = [tick, tick, tick]; // expect: WV-TS-003
  const merged = { ...snapshot }; // expect: WV-TS-002 + WV-TS-004
  const parsed = JSON.parse(RAW); // expect: WV-TS-007
  const copy = ROWS.map(normalize); // expect: WV-TS-009
  out[0] = scratch[0] + closure(1) + snapshot.bid + row[0] + merged.ask + parsed.a + copy[0];
}

function normalize(x: number): number {
  return x * 2;
}

export function coldPath(tick: number): number {
  const fresh = new Float64Array(8); // NOT hot: no finding
  const cb = (x: number) => x + tick; // NOT hot: no finding
  return fresh[0] + cb(1);
}

File: tests/verify/managed/fixtures/poisoned/poison.c
#include <stdlib.h>

/* poison.c — poisoned C fixture. Every commented rule id MUST fire. */

__attribute__((weft_hot))
static void on_frame(const unsigned char *pkt, struct ctx *ctx) {
    void *p = malloc(64);                        /* expect: WV-C-001 */
    struct hdr *q = calloc(1, 128);              /* expect: WV-C-002 */
    ctx->scratch = realloc(ctx->scratch, 256);   /* expect: WV-C-003 */
    char *dup = strdup("tick");                  /* expect: WV-C-005 */
    (void)p; (void)q; (void)dup;
    ctx->acc = (unsigned long)pkt[0];
}

static void cold_path(void) {
    void *p = malloc(64); /* NOT hot: no finding */
    (void)p;
}
